src/HatsEligibilitiesChain.sol

```diff
@@ -115,10 +116,10 @@ contract HatsEligibilitiesChain is HatsEligibilityModule {
       length = _getArgUint256(104 + clauseIndex * 32); // current clause length
 
       eligibleInClause = true;
-
+      nextClauseOffset += length * 20;
```

Comment:
can make this unchecked since the gas limit will be reached long before it ever overflows